sotrplib/handlers/basic.py

```diff
@@ -3,124 +3,137 @@
-        self.source_injector = source_injector or EmptySourceInjector()
-        self.source_catalogs = source_catalogs or []
-        self.preprocessors = preprocessors or []
-        self.postprocessors = postprocessors or []
-        self.forced_photometry = forced_photometry or EmptyForcedPhotometry()
-        self.source_subtractor = source_subtractor or EmptySourceSubtractor()
-        self.blind_search = blind_search or EmptyBlindSearch()
-        self.sifter = sifter or EmptySifter()
-        self.outputs = outputs or []
-
-        return
-
-    def run(self):
-        for map_idx, input_map in enumerate(self.maps):
-            input_map.build()
-
-            for preprocessor in self.preprocessors:
-                self.maps[map_idx] = preprocessor.preprocess(
-                    input_map=self.maps[map_idx]
-                )
-
-        # Generate sources based upon maximal bounding box of all maps
-        bbox = self.maps[0].bbox
-
-        for input_map in self.maps[1:]:
-            map_bbox = input_map.bbox
-            left = min(bbox[0].ra, map_bbox[0].ra)
-            bottom = min(bbox[0].dec, map_bbox[0].dec)
-            right = max(bbox[1].ra, map_bbox[1].ra)
-            top = max(bbox[1].dec, map_bbox[1].dec)
-            bbox = [SkyCoord(ra=left, dec=bottom), SkyCoord(ra=right, dec=top)]
-
-        all_simulated_sources = []
-
-        for simulator in self.source_simulators:
-            simulated_sources, catalog = simulator.generate(box=bbox)
-
-            all_simulated_sources.extend(simulated_sources)
-            self.source_catalogs.append(catalog)
-
-        for input_map in self.maps:
-            input_map.finalize()
-
-            input_map = self.source_injector.inject(
-                input_map=input_map, simulated_sources=all_simulated_sources
-            )
-
-            for postprocessor in self.postprocessors:
-                postprocessor.postprocess(input_map=input_map)
-
-            forced_photometry_candidates = self.forced_photometry.force(
-                input_map=input_map, catalogs=self.source_catalogs
-            )
-
-            source_subtracted_map = self.source_subtractor.subtract(
-                sources=forced_photometry_candidates, input_map=input_map
-            )
-
-            blind_sources, _ = self.blind_search.search(input_map=source_subtracted_map)
-
-            sifter_result = self.sifter.sift(
-                sources=blind_sources,
-                catalogs=self.source_catalogs,
-                input_map=source_subtracted_map,
-            )
-
-            for output in self.outputs:
-                output.output(
-                    forced_photometry_candidates=forced_photometry_candidates,
-                    sifter_result=sifter_result,
-                    input_map=input_map,
-                )
-
-        return
+import tempfile
+from typing import Any
+
+from typing_extensions import Self, TypeVar
+
+import pyinstrument
+
+from .base import BaseRunner
```

Comment:
I think this file is what ruff is complaining about. Make sure to ruff check --fix and ruff format.